pkg/beacon/analyzer.go

```diff
@@ -288,3 +303,121 @@ func countAndRemoveConsecutiveDuplicates(numberList []int64) ([]int64, map[int64
 	}
 	return result, counts
 }
+
+//getTsHistogramScore
+func getTsHistogramScore(min int64, max int64, tsList []int64) ([]int64, []int, map[int]int, float64) {
+
+	// get bucket list
+	// we currently look at a 24 hour period
+	bucketList := createBuckets(min, max, 24)
+
+	// use timestamps to get freqencies for buckets
+	freqList, freqCount, freqCV := createHistogram(bucketList, tsList)
+
+	// calculate first potential score
+	// histograms with bigger flat sections will score higher, up to 4 flat sections
+	// this will score well for graphs that have flat sections with a big distance between them,
+	// i.e. a beacon that alternates between 1 and 5 connections per hour
+	score1 := math.Ceil((float64(4)/float64(len(freqCount)))*1000) / 1000
+	if score1 > 1.0 {
+		score1 = 1.0
+	}
+
+	// calculate second potential score
+	// coefficient of variation will help score histograms that have jitter in the number of
+	// connections but where the overall graph would still look relatively flat and consistent
+	score2 := math.Ceil((1.0-float64(freqCV))*1000) / 1000
+	if score2 > 1.0 {
+		score2 = 1.0
+	}
+
+	return bucketList, freqList, freqCount, math.Max(score1, score2)
+
+}
+
+//createBuckets
+func createBuckets(min int64, max int64, total int64) []int64 {
+
+	// declare list
+	bucketList := make([]int64, total)
+
+	// calculate step size
+	step := (max - min) / (total - 1)
+
+	// set first bucket value to min timestamp
+	bucketList[0] = min
+
+	// create evenly spaced timestamp buckets
+	for i := int64(1); i < total; i++ {
+		bucketList[i] = min + (i * step)
+	}
+
+	// set first bucket value to max timestamp
+	bucketList[total-1] = max
+
+	return bucketList
+}
+
+func createHistogram(bucketList []int64, tsList []int64) ([]int, map[int]int, float64) {
+
+	i := 0
+	bucket := bucketList[i+1]
+	freqList := make([]int, len(bucketList)-1)
+
+	// loop over sorted timestamp list
+	for _, entry := range tsList {
+
+		// increment if still in the current bucket
+		if entry < bucket {
+			freqList[i]++
+			continue
+		}
+
+		// find the next bucket this value will fall under
+		for j := i + 1; j < len(bucketList)-1; j++ {
+			if entry < bucketList[j+1] {
+				i = j
+				bucket = bucketList[j+1]
+				break
+			}
+		}
+
+		// increment count
+		// this will also capture and increment for a situation where the final timestamp is
+		// equal to the final bucket
+		freqList[i]++
+	}
+
+	// make a fequency count map
+	freqCount := make(map[int]int)
```

Comment:
Please add a comment that `freqCount` is how often each value in `freqList` appears. Or, the frequency of the connection rates.